src/services/api.ts

```diff
@@ -48,20 +47,15 @@ export const getToken = async (
   if (!token) {
     throw new ValidationError('Required parameter "token" is missing.')
   }
-  try {
-    const response = await request<Token>(
-      `${config.get().apiUrl}/token?${new URLSearchParams({
-        chain,
-        token,
-      } as Record<string, string>)}`,
-      {
-        signal: options?.signal,
-      }
-    )
-    return response
-  } catch (e) {
-    throw await parseBackendError(e)
```

Comment:
parseBackendError as basically do two main things - using response.json() to access the response body message and then also creating an error with sets of error codes and error types dependant on status code of the response.

Both these tasks have been moved into the HTTPError class - with the async method to process the response.json() being triggered in the request function. The try/catch shouldn't be needed here anymore.